src/_gettsim/functions/policy_function.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import annotations
+
+import functools
+import inspect
+from collections.abc import Callable
+from datetime import date
+from typing import Any, TypeVar
+
+import numpy
+
+T = TypeVar("T")
+
+
+class PolicyFunction(Callable):
+    """
+    A function that computes an output vector based on some input vectors and/or
+    parameters.
+
+    Parameters
+    ----------
+    function:
+        The function to wrap. Argument values of the `@policy_info` are reused unless
+        explicitly overwritten.
+    module_name:
+        The name of the module where the function is defined.
+    function_name:
+        The name of the function in the DAG
+    start_date:
+        The date from which the function is active (inclusive).
+    end_date:
+        The date until which the function is active (inclusive).
+    params_key_for_rounding:
+        The key in the params dictionary that should be used for rounding.
+    skip_vectorization:
+        Whether the function should be vectorized.
+    """
+
+    def __init__(  # noqa: PLR0913
+        self,
+        function: Callable,
+        *,
+        module_name: str = "",
+        function_name: str | None = None,
+        start_date: date | None = None,
+        end_date: date | None = None,
+        params_key_for_rounding: str | None = None,
+        skip_vectorization: bool | None = None,
+    ):
+        info: dict[str, Any] = getattr(function, "__info__", {})
+
+        self.skip_vectorization: bool = _first_not_none_or_none(
+            skip_vectorization, info.get("skip_vectorization"), False
+        )
+
+        self.function = (
+            function if self.skip_vectorization else _vectorize_func(function)
+        )
+        self.module_name = module_name
+
+        self.name_in_dag: str = _first_not_none_or_none(
```

Comment:
I'd prefer two versions: `_first_not_none` for the first three use cases (just replace the `return None` by raising an error) and `_first_not_none_or_none` for the rounding case. Took me a while to figure out what exactly is happening here.